python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -333,6 +361,30 @@ def _get_num_gpus(
     return None
 
 
+def _get_num_tpus(
+    ray_start_params: Dict[str, str],
+    k8s_resource_limits: Dict[str, Any],
+    group_name: str,
+) -> Optional[int]:
+    """Get TPU custom resource annotation from ray_start_params or k8s_resource_limits,
+    with priority for ray_start_params.
+    """
+
+    if "TPU" in ray_start_params:
```

Comment:
In my understanding, `ray start` CLI doesn't have an option `--TPU`.

https://sourcegraph.com/github.com/ray-project/ray@cd3d3b7f8908e8adc821f1a79db5690150526ee7/-/blob/python/ray/scripts/scripts.py?L568

Reply:
I tried

```
> ray start --head --TPU=4
2024-05-21 23:10:31,470 - INFO - Note: NumExpr detected 32 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
2024-05-21 23:10:31,471 - INFO - NumExpr defaulting to 8 threads.
Usage: ray start [OPTIONS]
Try 'ray start --help' for help.

Error: No such option: --TPU
```

Reply:
There was a time when we passed in custom resources like:
`resources: { "TPU": 4}` 
in the Ray start parameters. These should not be needed anymore since we should rely on Ray to detect the available devices.